addons/binding/org.openhab.binding.amazondashbutton/src/main/java/org/openhab/binding/amazondashbutton/internal/discovery/AmazonDashButtonDiscoveryService.java

```diff
@@ -0,0 +1,105 @@
+package org.openhab.binding.amazondashbutton.internal.discovery;
+
+import static org.openhab.binding.amazondashbutton.AmazonDashButtonBindingConstants.*;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.amazondashbutton.internal.ArpRequestTracker;
+import org.openhab.binding.amazondashbutton.internal.ArpRequestTracker.ArpRequestHandler;
+import org.openhab.binding.amazondashbutton.internal.PcapUtil;
+import org.pcap4j.core.PcapNetworkInterface;
+import org.pcap4j.packet.ArpPacket;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AmazonDashButtonDiscoveryService} is responsible for discovering Amazon Dash Buttons. It does so by
+ * capturing ARP requests from all available network devices.
+ *
+ * While scanning the user has to press the button in order to send an ARP request packet. The
+ * {@link AmazonDashButtonDiscoveryService} captures this packet and checks the device'sMAC address which sent the
+ * request against a static list of vendor prefixes ({@link #isAmazonVendor(String)}).
+ *
+ * If an Amazon MAC address is detected a {@link DiscoveryResult} is built and passed to
+ * {@link #thingDiscovered(DiscoveryResult)}.
+ *
+ * @author Oliver Libutzki - Initial contribution
+ *
+ */
+public class AmazonDashButtonDiscoveryService extends AbstractDiscoveryService {
+
+    private static final int DISCOVER_TIMEOUT_SECONDS = 30;
+    @SuppressWarnings("unused")
+    private static final Logger logger = LoggerFactory.getLogger(AmazonDashButtonDiscoveryService.class);
+
+    private List<ArpRequestTracker> arpRequestListeners = new CopyOnWriteArrayList<ArpRequestTracker>();
+
+    public AmazonDashButtonDiscoveryService() {
+        super(Collections.singleton(DASH_BUTTON_THING_TYPE), DISCOVER_TIMEOUT_SECONDS, false);
+    }
+
+    @Override
+    protected void startScan() {
```

Comment:
Yes, you are right. The problem has been reported in #1235 as well. Unfortunately I just have tested the binding on my Windows machine so far. I will give it a try on my RaspPi.